symengine/matrix.h

```diff
@@ -118,90 +118,88 @@ class DenseMatrix : public MatrixBase
     // and throw an exception if a method is not applicable.
 
     // Get and set elements
-    virtual RCP<const Basic> get(unsigned i, unsigned j) const;
-    virtual void set(unsigned i, unsigned j, const RCP<const Basic> &e);
-    virtual vec_basic as_vec_basic() const;
+    RCP<const Basic> get(unsigned i, unsigned j) const override;
+    void set(unsigned i, unsigned j, const RCP<const Basic> &e) override;
+    vec_basic as_vec_basic() const;
 
-    virtual unsigned nrows() const
+    unsigned nrows() const override
     {
         return row_;
     }
-    virtual unsigned ncols() const
+    unsigned ncols() const override
     {
         return col_;
     }
 
-    virtual bool is_lower() const;
-    virtual bool is_upper() const;
-    virtual tribool is_zero() const;
-    virtual tribool is_diagonal() const;
-    virtual tribool is_real(const Assumptions *assumptions
-                            = nullptr) const override;
-    virtual tribool is_symmetric() const;
-    virtual tribool is_hermitian() const;
-    virtual tribool is_weakly_diagonally_dominant() const;
-    virtual tribool is_strictly_diagonally_dominant() const;
-    virtual tribool is_positive_definite() const;
-    virtual tribool is_negative_definite() const;
+    bool is_lower() const;
+    bool is_upper() const;
+    tribool is_zero() const;
+    tribool is_diagonal() const;
+    tribool is_real(const Assumptions *assumptions = nullptr) const;
+    tribool is_symmetric() const;
+    tribool is_hermitian() const;
+    tribool is_weakly_diagonally_dominant() const;
+    tribool is_strictly_diagonally_dominant() const;
+    tribool is_positive_definite() const;
+    tribool is_negative_definite() const;
 
```

Comment:
Same question applies for these methods.

Reply:
Same as above.

As a side note: I now think these would be better implemented using double dispatch as in #1918. This way we decouple the classes from the functions on the classes. Couldn't be easily done before since `Matrix` does not inherit from `Basic` which the proposed `ImmutableDenseMatrix` in the PR does.